REVIEW_COPY.md

```diff
@@ -0,0 +1,312 @@
+
+That's a pretty sweeping statement for a broad term like 'drag & drop', so I'd better back it up! To do so I'll show you three examples of drag & drop UIs built without draggables or dropzones. These examples come from real projects, but I've stripped them down to their drag & drop essentials for this post. I'll include what I believe to be the most relevant code snippets in those post, and invite you to look at [the full source code] if you'd like to learn more!
+
+Before we start with those examples, lets start with a quick refresher on draggables and dropzones.
+
+## A Refresher on Draggables & Dropzones
+
+To learn what draggables and dropzones are lets look at [Trello], an application where they are easy to identify. In Trello you can drag cards between lists. This makes cards draggables and lists dropzones.
+
+![A screenshot of the Trello app, showing some lists and cards][trello.png]
+
+On closer inspection though things aren't as clear cut. For example, when you drop a card next to a dropzone you might expect that card to return to its original list, or even get lost. But what happens is that the card will land in the list nearest to where we dropped it. That's much nicer for the user, but it's not clear how we might achieve it using draggables and dropzones.
+
+## Our first example, A Timeslot Selector
+
+Our first example is a UI for selecting a time slot in a day. This UI represents a day as a horizontal bar, in which you can select a slot using drag & drop.
+
+![Selecting a time slot by drag & drop][slider.gif]
+
+I'd like to start these examples by imagining what a draggables and dropzones implementation might look like. In this example it's unclear what our draggables and dropzones even are. We could make the time slider the draggable, because clicking anywhere in it should start the drag operation even if we don't drag the time slider itself. The screen as a whole might serve as a dropzone, because the user should be able to release the cursor anywhere to finish the drag. Already this approach feels pretty hacky, and we haven't even written any code yet!
+
+Lets start again from scratch without assuming draggables or dropzones this time and see where it takes us. We'll begin in a way we often begin an Elm application: by designing our Model. Our application needs to store a time slot which is nothing more than a start and end hour.
+
+```elm
+-- In this example a time slot always covers whole hours,
+-- but we could make this minutes or seconds if we wanted.
+type alias Model =
+    { from : Hour, until : Hour }
+
+type alias Hour =
+    Int
+```
+
+Now for the drag & drop behavior. In our example all you can do is drag left-to-right. This means that when the user presses down we store the hour the cursor is over as the left bound of the time slot. Then as the cursor moves to the right we will update the `until` time of the timeslot, until the user releases.
```

Comment:
does this only work left-to-right or does it also work right-to-left?

Reply:
Only left-to-right. I'm on the fence about this one. Making the code support right-to-left dragging as well isn't a huge addition, but it is additional logic that distracts from the 'drag & drop plumbing' that I'm trying to focus on. In this version I gave precedent to the latter.

What do you feel is the best move here? I'd like to examples to be as light-weight as possible, but also realistic enough that they demonstrate the approach to drag & drop I'm championing here can be used in real world applications.

Reply:
it seems like you're trying to demonstrate that

a) drag and drop is not always appropriate
b) beacons are sometimes more appropriate

so saying "in the beacons version it is trivial to add additional behavior" seems fine.

Reply:
Makes sense! I updated the code so you can select in both directions.